src/main/java/frc/lib/io/motor/MotorIORev.java

```
@@ -0,0 +1,236 @@
+     */
+
+    /**
+     * Maps MotorIO.PIDSlot to REV ClosedLoopSlot.
+     */
+    private static final EnumMap<MotorIO.PIDSlot, ClosedLoopSlot> SLOT_MAP =
+        new EnumMap<>(MotorIO.PIDSlot.class);
+    static {
+        SLOT_MAP.put(MotorIO.PIDSlot.SLOT_0, ClosedLoopSlot.kSlot0);
+        SLOT_MAP.put(MotorIO.PIDSlot.SLOT_1, ClosedLoopSlot.kSlot1);
+        SLOT_MAP.put(MotorIO.PIDSlot.SLOT_2, ClosedLoopSlot.kSlot2);
+    }
+
+    /**
+     * Gets the corresponding REV ClosedLoopSlot for a given MotorIO.PIDSlot.
+     * 
+     * @param slot The MotorIO.PIDSlot to convert.
+     */
+    protected ClosedLoopSlot getClosedLoopSlot(MotorIO.PIDSlot slot)
+    {
+        return SLOT_MAP.getOrDefault(slot, ClosedLoopSlot.kSlot0);
+    }
+
+    /**
+     * Maps MotorIO.ControlType to REV ControlType.
+     */
+    private static final EnumMap<ControlType, com.revrobotics.spark.SparkBase.ControlType> CONTROL_TYPE_MAP =
+        new EnumMap<>(ControlType.class);
+    static {
+        CONTROL_TYPE_MAP.put(ControlType.VOLTAGE,
+            com.revrobotics.spark.SparkBase.ControlType.kVoltage);
+        CONTROL_TYPE_MAP.put(ControlType.CURRENT,
+            com.revrobotics.spark.SparkBase.ControlType.kCurrent);
+        CONTROL_TYPE_MAP.put(ControlType.DUTYCYCLE,
+            com.revrobotics.spark.SparkBase.ControlType.kDutyCycle);
+        CONTROL_TYPE_MAP.put(ControlType.POSITION,
+            com.revrobotics.spark.SparkBase.ControlType.kPosition);
+        CONTROL_TYPE_MAP.put(ControlType.VELOCITY,
+            com.revrobotics.spark.SparkBase.ControlType.kVelocity);
+    }
+
+    /**
+     * Gets the corresponding REV ControlType for a given MotorIO.ControlType.
+     * 
+     * @param type The MotorIO.ControlType to convert.
+     */
+    protected com.revrobotics.spark.SparkBase.ControlType getRevControlType(ControlType type)
+    {
+        return CONTROL_TYPE_MAP.getOrDefault(type,
+            com.revrobotics.spark.SparkBase.ControlType.kDutyCycle);
+    }
```

These enums should map 1-1, this should throw an IllegalStateException rather than silently defaulting